app/platform/__init__.py

```diff
@@ -137,3 +147,229 @@ def edit_collection(grant_id: UUID4, collection_id: UUID4) -> ResponseReturnValu
+
+@platform_blueprint.route(
+    "/grants/<uuid:grant_id>/developers/collections/<uuid:collection_id>/sections/<uuid:section_id>/forms/add",
+    methods=["GET", "POST"],
+)
+@mhclg_login_required
+@auto_commit_after_request
+def add_form(grant_id: UUID4, collection_id: UUID4, section_id: UUID4) -> ResponseReturnValue:
+    section = get_section_by_id(section_id)
+    form_type = request.args.get("form_type", None)
+    if not form_type:
+        return render_template(
+            "platform/developers/select_form_type.html",
+            grant=section.collection_schema.grant,
+            collection=section.collection_schema,
+            section=section,
+        )
+
+    form = FormForm()
+    if form.validate_on_submit():
+        try:
+            assert form.title.data is not None
+            create_form(title=form.title.data, section=section)
+            return redirect(
+                url_for(
+                    "platform.manage_section", grant_id=grant_id, collection_id=collection_id, section_id=section_id
+                )
+            )
+        except DuplicateValueError as e:
+            field_with_error: Field = getattr(form, e.field_name)
+            field_with_error.errors.append(f"{field_with_error.label.text} already in use")  # type:ignore[attr-defined]
+    return render_template(
+        "platform/developers/add_form.html",
+        grant=section.collection_schema.grant,
+        collection=section.collection_schema,
+        section=section,
+        form_type=form_type,
+        form=form,
+    )
+
+
+@platform_blueprint.route(
+    "/grants/<uuid:grant_id>/developers/collections/<uuid:collection_id>/sections/<uuid:section_id>/forms/<uuid:form_id>/edit",
+    methods=["GET", "POST"],
+)
+@mhclg_login_required
+@auto_commit_after_request
+def edit_form(grant_id: UUID4, collection_id: UUID4, section_id: UUID4, form_id: UUID4) -> ResponseReturnValue:
+    form = get_form_by_id(form_id)
+    wt_form = FormForm(obj=form)
+    if wt_form.validate_on_submit():
+        try:
+            assert wt_form.title.data is not None
+            update_form(form=form, title=wt_form.title.data)
+            return redirect(
+                url_for(
+                    "platform.manage_form",
+                    grant_id=grant_id,
+                    collection_id=collection_id,
+                    section_id=section_id,
+                    form_id=form_id,
+                    back_link="manage_section",
+                )
+            )
+        except DuplicateValueError as e:
+            field_with_error: Field = getattr(wt_form, e.field_name)
+            field_with_error.errors.append(f"{field_with_error.label.text} already in use")  # type:ignore[attr-defined]
+
+    return render_template(
+        "platform/developers/edit_form.html",
+        grant=form.section.collection_schema.grant,
+        collection=form.section.collection_schema,
+        section=form.section,
+        form=form,
+        wt_form=wt_form,
```

Comment:
This breaks error summaries on this page - the templates handle error summary 'magic' by expecting the wtform to be passed in under the `form` variable name.

A bit annoying that the magic here is working against us, but it otherwise is quite helpful so can we maybe:

- rename `form` to `form_obj` or `db_form` or whatever, and `wt_form` to `form` - passing that into the template accordingly
- call out Form DB table something else - that's probably a bigger one to think about later 👀

Reply:
Yeah let's park the name of the table, we can tackle the `FormForm` class at the same time!